### PR TITLE
Update Pulumi codegen to net6.0

### PR DESCRIPTION
**Pull request by Frassle**
# Description

.NET Core 3.1 is out of support on December 13th:
https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core

We've already stopped testing on .NET Core 3.1, and program gen has been targeting only 6.0 for a while now as well.

This updates sdkgen to only 6.0 as well.

We'll need to ensure the version number in pkg/codegen/dotnet/gen.go matches the version number that https://github.com/pulumi/pulumi-dotnet/pull/10 release as.

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works

- [x] I have run `make changelog` and committed the `changelog/pending/<file>` documenting my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Service API version

### CODE REVIEW

**Comment by mikhailshilkov** [comment]
What are the benefits of doing it and downsides of keeping 3.1? Just curious.

**Comment by Frassle** [comment]
> What are the benefits of doing it and downsides of keeping 3.1? 

Benefit is if anyone is still targeting 3.1 they could keep building against Pulumi.

Downside is it limits the api area of dotnet we can use (there's quite a few useful things added in 6.0).
We're also not testing or supporting 3.1 (and neither is Microsoft).

**Comment by mikhailshilkov** [comment]
> there's quite a few useful things added in 6.0

That's interesting. Any concrete enhancements that you have in mind that we can apply now?

File: pkg/codegen/dotnet/gen.go [hunk 1]
```diff
@@ -2076,7 +2076,7 @@ func genProjectFile(pkg *schema.Package,
 		// only add a package reference to Pulumi if we're not referencing a local Pulumi project
 		// which we usually do when testing schemas locally
 		if !referencedLocalPulumiProject {
-			packageReferences["Pulumi"] = "[3.23.0,4)"
+			packageReferences["Pulumi"] = "[3.51.0,4)"
```
Comment (kpitzen) [inline pkg/codegen/dotnet/gen.go:2079]: What does this actually do?  Are we checking for the currently released version?

**Review by kpitzen — APPROVED** [review]
Approved pending my one question.

File: pkg/codegen/dotnet/gen.go (hunk 1 above)
Comment (Frassle) [reply pkg/codegen/dotnet/gen.go:2079]: This tells the SDK to use at least Pulumi.nuget 3.51 (which targets net6). I'm not sure it's strictly needed, I think a net6 library can reference a net3.1 library, I can check that quickly.

Comment (Frassle) [reply pkg/codegen/dotnet/gen.go:2079]: Yeh actually this seems to work, so we could probably drop this.

**Comment by Frassle** [comment]
bors merge

**Pull request merged**